openapis/swagger.json

```diff
@@ -1,594 +1,794 @@
+                    }
+                ]
+            },
+            "delete": {
+                "tags": [
+                    "index"
+                ],
+                "summary": "Delete an entry from the index",
+                "description": "",
+                "operationId": "deleteEntry",
+                "produces": [
+                    "application/json"
+                ],
+                "parameters": [
+                    {
+                        "name": "UUID",
+                        "in": "path",
+                        "description": "entry id",
+                        "required": true,
+                        "type": "string"
+                    },
+                    {
+                        "name": "REV",
+                        "in": "path",
+                        "description": "data revision - simple consistency mechanism",
+                        "required": true,
+                        "type": "string"
+                    }
+                ],
+                "responses": {
+                    "400": {
+                        "description": "Invalid ID supplied"
+                    }
+                },
+                "security": [
+                    {
+                        "cdis_auth": [
+                            "write:index",
+                            "read:index"
+                        ]
+                    }
+                ]
             }
-          },
-          "400": {
-            "description": "Invalid input"
-          }
         },
-        "security": [
-          {
-            "cdis_auth": [
-              "write:index",
-              "read:index"
-            ]
-          }
-        ]
-      }
-    },
-    "/alias/{ALIASSTRING}?rev={REV}": {
-      "put": {
-        "tags": [
-          "alias"
-        ],
-        "summary": "Update an alias",
-        "description": "",
-        "operationId": "updateAlias",
-        "consumes": [
-          "application/json"
-        ],
-        "produces": [
-          "application/json"
-        ],
-        "parameters": [
-          {
-            "name": "ALIASSTRING",
-            "in": "path",
-            "type": "string",
-            "required": true,
-            "description": "alias like ark:/31807/DC1-TESTARK"
-          },
-          {
-            "name": "REV",
-            "in": "path",
-            "type": "string",
-            "pattern": "^[a-z,0-9]{8}$",
-            "required": true
-          },
-          {
-            "in": "body",
-            "name": "body",
-            "description": "Metadata object that needs to be added to the store",
-            "required": true,
-            "schema": {
-              "$ref": "#/definitions/AliasInputInfo"
-            }
-          }
-        ],
-        "responses": {
-          "200": {
-            "description": "successful operation",
-            "schema": {
-              "$ref": "#/definitions/AliasRef"
+        "/index/version/{UUID}": {
```

Comment:
I think it would be better to not overload the second subdirectory to be either {UUID} or `version`, how about just update the existing index creation endpoint to allow an optional `base_id` property in the json payload?

Reply:
Since the base_id can be NULL, how indexd can create a new version of the index record? If we want to update the existing index creation endpoint with optinal base_id in the json payload, all the index record should have non-NULL base_id